BridgeAppSDK/SBAGenericStepDataSource.swift

```diff
@@ -0,0 +1,614 @@
+        
+        // inform delegate that answers have changed
+        if let delegate = delegate {
+            delegate.answersDidChange()
+        }
+    }
+    
+    /**
+     Select or deselect the answer option for a specific IndexPath.
+     @param   indexPath   The IndexPath that represents the TableItemGroup in the tableView
+     */
+    open func selectAnswer(selected: Bool, at indexPath: IndexPath) {
+        
+        let itemGroup = self.itemGroup(at: indexPath)
+        itemGroup?.select(selected, indexPath: indexPath)
+        
+        // inform delegate that answers have changed
+        if let delegate = delegate {
+            delegate.answersDidChange()
+        }
+    }
+    
+    /**
+     Retrieve the current ORKStepResult.
+     @return    An ORKStepResult object with the current answers for all of its ORKFormItems
+     */
+    open func results(parentResult: ORKStepResult) -> ORKStepResult {
+        
+        if let formItems = formItemsWithAnswerFormat() {
+            
+            // "Now" is the end time of the result, which is either actually now,
+            // or the last time we were in the responder chain.
+            
+            let now = parentResult.endDate
+            var qResults: Array<ORKResult> = Array()
+            
+            for formItem: ORKFormItem in formItems {
+                
+                
+                // Skipped forms report a "null" value for every item -- by skipping, the user has explicitly said they don't want
+                // to report any values from this form.
+                
+                var answer = ORKNullAnswerValue()
+                var answerDate = now
+                var systemCalendar = Calendar.current
+                var systemTimeZone = NSTimeZone.system
+                
+                // TODO: Josh Bruhin, 6/12/17 - implement skipped?
+                let skipped = false
+                if !skipped {
+                    
+                    if let itemGroup = itemGroup(with: formItem.identifier) {
+                        answer = itemGroup.answer
+                        answerDate = itemGroup.answerDate ?? now
+                        systemCalendar = itemGroup.calendar
+                        systemTimeZone = itemGroup.timezone
+                    }
+                }
+                
+                let result = formItem.answerFormat?.result(withIdentifier: formItem.identifier, answer: answer)
+                let impliedAnswerFormat = formItem.answerFormat?.implied()
+                
+                if let dateAnswerFormat = impliedAnswerFormat as? ORKDateAnswerFormat, let dateQuestionResult = result as? ORKDateQuestionResult {
+                    if let _ = dateQuestionResult.dateAnswer {
+                        let usedCalendar = dateAnswerFormat.calendar ?? systemCalendar
+                        dateQuestionResult.calendar = usedCalendar
+                        dateQuestionResult.timeZone = systemTimeZone
```

Comment:
Copied directly form RK - (ORKStepResult *)result in ORKFormStepVC. I'll look into it...